api/migrations/versions/dc2b619f65a9_add_archive_table_and_triggers.py

```diff
@@ -0,0 +1,126 @@
+"""Add archive table and triggers
+
+Revision ID: dc2b619f65a9
+Revises: 72e6af46be22
+Create Date: 2023-11-14 11:17:57.848580
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+from alembic_utils.pg_function import PGFunction
+from alembic_utils.pg_trigger import PGTrigger
+
+
+# revision identifiers, used by Alembic.
+revision = 'dc2b619f65a9'
+down_revision = '72e6af46be22'
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.create_table('archive',
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('table_name', sa.VARCHAR(length=100), nullable=True),
+    sa.Column('record_type', sa.VARCHAR(length=100), nullable=True),
+    sa.Column('record_id', sa.INTEGER(), nullable=True),
+    sa.Column('operation', sa.VARCHAR(length=50), nullable=True),
+    sa.Column('old_values', postgresql.JSON(astext_type=sa.Text()), nullable=True),
+    sa.Column('new_values', postgresql.JSON(astext_type=sa.Text()), nullable=True),
+    sa.Column('most_recent', sa.BOOLEAN(), nullable=True),
+    sa.Column('recorded_at', postgresql.TIMESTAMP(), nullable=True))
```

Comment:
For some reason this migration was missing `sa.PrimaryKeyConstraint('id')`

```suggestion
    sa.Column('recorded_at', postgresql.TIMESTAMP(), nullable=True)
    sa.PrimaryKeyConstraint('id')
    )
```

I think we also need to add the model to `base.py`:

```diff
diff --git a/api/db/base.py b/api/db/base.py
index 897846c..2066187 100644
--- a/api/db/base.py
+++ b/api/db/base.py
@@ -1,4 +1,5 @@
 from db.base_class import Base
+from models.archive import *
```

Alembic only detected this table creation after did this.